### PR TITLE
Swap MatAccordion to MatTree in Tables List component

### LINKED ISSUE #122: Refactor: Change MatAccordion to MatTree in Tables List component
## Which files are to refactor

Replace `MatAccordion` with `MatTree` component in `Tables List` to mach desired visual look and behavior.

## Reason of refactoring

Missed it during code review.

### PR DESCRIPTION
**Pull request by Lukash88**
## Summary

Change MatAccordion to MatTree in Tables List component.

## Control List

- [x] Code formatting checked,
- [x] Coding Guidelines compliance checked,
- [ ] Documentation updated (if necessary),
- [ ] Automated tests updated (if necessary),

## Screenshots

Tables List expanded:
image

Tables List collapsed:
image

### CODE REVIEW

**Review by StraykerPL — CHANGES_REQUESTED** [review]
Change MatNestedTree to MatTree with flat nodes, refactor styles to make them more readable, check code formatting once more.

File: binder-web-frontend/src/api/model/tableFlatNode.ts
Comment (StraykerPL) [inline binder-web-frontend/src/api/model/tableFlatNode.ts:1]: If this is your code, not generated, move to `shared/models` folder.

File: binder-web-frontend/src/api/model/tableNode.ts
Comment (StraykerPL) [inline binder-web-frontend/src/api/model/tableNode.ts:1]: Same here.

File: binder-web-frontend/src/pages/home/components/tables-list/tables-list.component.scss [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
-      
-      &-content {
-        margin-left: 2rem;
+    }
 
-        &:hover {
-          cursor: pointer;
-        }
+    &-header {
+      display: inline-flex;
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/home/components/tables-list/tables-list.component.scss]: Change to `flex` only.

File: binder-web-frontend/src/pages/pages.module.ts [hunk 2]
```diff
@@ -24,7 +35,7 @@ import { MainSpaceComponent } from './home/components/main-space/main-space.comp
     DomainButtonsComponent,
     HeaderComponent,
     SidePanelComponent,
-    MainSpaceComponent
+    MainSpaceComponent
```
Comment (StraykerPL) [inline binder-web-frontend/src/pages/pages.module.ts]: Clean spaces.

**Review by StraykerPL — CHANGES_REQUESTED** [review]

File: binder-web-frontend/src/api/model/tableFlatNode.ts
Comment (Lukash88) [reply binder-web-frontend/src/api/model/tableFlatNode.ts:1]: Resolved 355634b.

File: binder-web-frontend/src/api/model/tableNode.ts
Comment (Lukash88) [reply binder-web-frontend/src/api/model/tableNode.ts:1]: Resolved 355634b.

File: binder-web-frontend/src/pages/home/components/tables-list/tables-list.component.scss (hunk 1 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/home/components/tables-list/tables-list.component.scss]: Resolved 355634b.

File: binder-web-frontend/src/pages/pages.module.ts (hunk 2 above)
Comment (Lukash88) [reply binder-web-frontend/src/pages/pages.module.ts]: Resolved 355634b.

**Review by StraykerPL — DISMISSED** [review]

**Review by StraykerPL — APPROVED** [review]

**Pull request merged**